src/odemis/gui/cont/tabs.py

```diff
@@ -508,20 +508,30 @@ def load_data(self, data):
         # Create streams from data
         streams = data_to_static_streams(data)
 
-        # TODO: Clear previous overview streams
-        self.clear_data()
-
         for s in streams:
+            self.clear_data(s)
             scont = self._overview_stream_controller.addStream(s, add_to_view=True)
             scont.stream_panel.show_remove_btn(True)
 
-    def clear_data(self):
+    def clear_data(self, new_stream=None):
         """
         Clear the tab data upon resetting the project:
         - Clear overview map streams
-        - Clear live streams data
-        """
-        self._overview_stream_controller.clear()
+        @:param new_stream: (StaticStream or None) the newly acquired overview stream
+        """
+        # Clear old static streams from the view
+        # TODO: Remove old streams from the panel as well, currently it's hidden
+        if new_stream:
+            # Replace only the stream matching the new stream in acquisition type and name
+            old_stream = next((stream for stream in self.tab_data_model.streams.value if isinstance(stream, StaticStream)
+                               and stream.acquisitionType.value == new_stream.acquisitionType.value and stream.name.value == new_stream.name.value), None)
+            if old_stream:
+                self._overview_stream_controller.removeStream(old_stream)
```

Comment:
You could make it a little simpler, by merging it into the "all streams" code. Assuming that you just are filtering on the name, it'd look like this (but if you pass the stream, it's very similar):
```python
static_streams = [stream for stream in self.tab_data_model.streams.value if isinstance(stream, StaticStream)]
for stream in static_streams:
    if filter and stream.name.value != filter:
        continue
    self._overview_stream_controller.removeStream(stream)
```
Note that it'd remove all the streams matching the filter. I think this is fine. Typically, you'd have only one stream with the same name.